File: src/modules/presentation/io/tripled/social/client/presentation/request/Visitor.java
package io.tripled.social.client.presentation.request;

import io.tripled.social.client.presentation.Output;
import io.tripled.social.client.presentation.request.FollowCliRequest;
import io.tripled.social.client.presentation.request.PostCliRequest;
import io.tripled.social.client.presentation.request.ReadCliRequest;
import io.tripled.social.client.presentation.request.WallCliRequest;

public interface Visitor {

  void visit(FollowCliRequest request, Output output);

  void visit(PostCliRequest request, Output output);

  void visit(ReadCliRequest request, Output output);

  void visit(WallCliRequest request, Output output);
}

File: src/main/kotlin/io/tripled/social/client/presentation/request/Visitor.kt
package io.tripled.social.client.presentation.request

import io.tripled.social.client.presentation.Output

interface Visitor {
    fun visit(request: FollowCliRequest, output: Output)

    fun visit(request: PostCliRequest, output: Output)

    fun visit(request: ReadCliRequest, output: Output)

    fun visit(request: WallCliRequest, output: Output)
}
